### ISSUE TITLE
Backpropogating ScalePyramid is slow

### ISSUE THREAD

**Issue by connorlee77** [kornia/kornia#431]
I have a ResnetFCN with its output fed through the ScalePyramid (default parameters). Is there a reason in the implementation of [ScalePyramid ](https://github.com/kornia/kornia/blob/master/kornia/geometry/transform/pyramid.py) for why backpropagation may be very slow?

**Comment by ducha-aiki** [comment]
You probably don't need ScalePyramid for ResNetFCN, as it samples scales too dense - it is designed for local features. Try build_pyramid https://github.com/kornia/kornia/blob/master/kornia/geometry/transform/pyramid.py#L248   instead . Second, the bluring is slow but essential part. We are planning to speed-up this, but not yet

**Comment by connorlee77** [comment]
Are the blurring convolutions the slow part of scalepyramid or is it creating the pyramids?

**Comment by ducha-aiki** [comment]
Good question. Would you like to help us benchmark it? :)

**Comment by connorlee77** [comment]
The latency was actually due to a non-optimal correlation function in my loss function. Thanks for the help!
